python/mlc_chat/compiler/model/llama_quantization.py

```diff
@@ -2,100 +2,25 @@
 Quantization specs for Llama2 architecture.
 TODO: add docstring
 """
-from typing import Callable, Dict, List, Optional
+from typing import Tuple
 
-import tvm
-from tvm.runtime import NDArray
+from tvm.relax.frontend import nn
 
 from ..parameter import QuantizeMapping
-from ..quantization import QuantizeConfig
-from ..quantization.group_quantizer import te_quantize as te_group_quantize
-from .llama_config import LlamaConfig
+from ..quantization import GroupQuantizeConfig
 from .llama_model import LlamaForCasualLM
 
 
-def huggingface_group_quantize(
-    model_config: LlamaConfig,
-    quantize_config: QuantizeConfig,
-    target: Optional[tvm.target.Target] = None,
-) -> QuantizeMapping:
-    """Returns a parameter mapping that maps a parameter in MLC LLM's model
-    definition to its eventual names and values after quantization.
-
-    Parameters
-    ----------
-    model_config : LlamaConfig
-        The configuration of the Llama model.
-    quantize_config : GroupQuantizeConfig
-        The configuration of the group quantization.
-    target : Optional[tvm.target.Target]
-        The target device to run the quantization on, by default None, which
-        means the quantization will be run on CPU.
-
-    Returns
-    -------
-    quantize_map : QuantizeMapping
-        The parameter mapping from a parameter in MLC LLM's model definition to
-        its eventual names and values after quantization.
-    """
-
-    def group_quantize(
-        param: NDArray, config: QuantizeConfig, target: Optional[tvm.target.Target] = None
-    ):
-        if target is None or target.kind.name == "llvm":
-            target = tvm.target.Target("llvm")
-            device = tvm.cpu()
-        elif target.kind.name == "cuda":
-            device = tvm.cuda()
-        else:
-            raise ValueError(f"Invalid target device: {target}")
-        param_tensor = tvm.te.placeholder(param.shape, dtype=param.dtype, name="param")
-        weight_compute, scale_compute, other_computes = te_group_quantize(  # type: ignore
-            param_tensor, config
-        )
-        s = tvm.te.create_schedule(  # pylint: disable=invalid-name
-            [compute.op for compute in [weight_compute, scale_compute] + other_computes]
-        )
-        if target.kind.name == "cuda":
-            # thread_binding for cuda
-            for compute in [weight_compute, scale_compute] + other_computes:
-                xo, xi = s[compute].split(compute.op.axis[0], 256)  # pylint: disable=invalid-name
-                s[compute].bind(xo, tvm.te.thread_axis("blockIdx.x"))
-                s[compute].bind(xi, tvm.te.thread_axis("threadIdx.x"))
-        f_quantize = tvm.build(
-            s, [param_tensor, weight_compute, scale_compute], name="group_quantize", target=target
+def llama_group_quantization(
```

Comment:
IIUC we could simply apply the quantization to the entire model without having to do so layer-by-layer in this case (group quant)